### ISSUE TITLE
Converted files pause every few seconds - unwatchable

### ISSUE THREAD

**Issue by orinoco77** [mdhiggins/sickbeard_mp4_automator#555]
I've got the scripts installed and working, but the output isn't great and I'm not sure why. Every file I've converted has the same issue (all converted from mkvs). About once every 30 seconds or so the video will pause on the same frame for up to 5 seconds, and then start up again as if the last few seconds have been played correctly (so I'm basically missing about 5 seconds in every 35). 5 seconds doesn't sound like a lot, but it's surprising how much can happen. It basically makes the files unwatchable.

I'm using the default options for the video conversion. The only thing I've changed is the paths, so that they point to the right places.

At first I thought perhaps my android TV box wasn't handling the files very well, but I've checked them out with my macbook pro and they're just as bad there. The problem has been encoded into the files, it's not just in the playback.

Any ideas?

**Comment by noremacyug** [comment]
I'm sure you already have, but you did verify the .mkvs themselves are fine, right?

**Comment by orinoco77** [comment]
Yep. The original files are fine, or at least they play fine on my macbook pro (so the files themselves aren't an issue). The reason I wanted to try the scripts was that they tend to be jerky, or to buffer a lot, on my android box. I'm guessing this is because the mkv needs decoding in a way that mp4s don't (or at least the mkvs are more complex/difficult). Basically I noticed I was having problems specifically with mkv files that I didn't have with mp4s, and converting one to the other seemed like a sane and sensible fix, but actually the files I've ended up with are much worse than the originals :(

**Comment by mdhiggins** [comment]
What device are you playing these files back on?
What software is being used for playback?
What are your encoding settings?
Are you seeing this playback issues on programs like VLC as well?
Are the MKVs being transcoded or just remuxed?

If they're just being remuxed, it would be strange since the video codec and data for the video stream is essentially unchanged and shouldn't create much of an impact

Also please post your autoProcess.ini file and some logging from one of the conversions, and ideally send me a link to a sample of one of the videos you've created that's having the problems. That's the quickest way to try to debug this issue.

**Comment by orinoco77** [comment]
Sorry for the delay. Before I start, I should note that since I had problems with the first couple of files, I've had no further issues, even without making any changes. I don't know what that indicates, but I'm certainly pleased that things seem to be working great now. This project is awesome and I'm extremely glad it exists. I'm a developer myself, and if I hadn't found this I was planning to hack something together for my own use. Your way is much nicer :)

Ok, here's some more info:
Device: Keedox M8 Android box (Amlogic S802) running Android 4.4
Software: Kodi 16 (Jarvis)

I'm not sure which encoding settings you want, if it's for the conversion to mp4, I'd guess it's doing whatever the default is, since I've not changed a lot of options (given the speed at which it's done, it's probably just remuxing). If you want the settings the original videos were encoded with, I have no idea, I'm afraid, and I've since deleted them, which in hindsight was foolish :(

The same playback problems were evident in VLC.

As I said above, I'm pretty sure they're just being remuxed, but my autoProcess.ini file should clarify that, I guess. I'm attaching it here.
[autoProcess.zip](https://github.com/mdhiggins/sickbeard_mp4_automator/files/460072/autoProcess.zip)

This is the log from one of the dodgy files - just a note, I removed the move_to option when I figured out it didn't do quite what I wanted, so it's not present in the autoProcess.ini, but it used to move finished files to c:\sync-tmp.

2016-09-03 15:13:45 - SonarrPostConversion - INFO - Sonarr extra script post processing started.
2016-09-03 15:13:45 - readSettings - INFO - C:\Python27\python.exe
2016-09-03 15:13:45 - readSettings - WARNING - Audio bitrate was invalid, defaulting to 256 per channel.
2016-09-03 15:13:45 - SonarrPostConversion - INFO - Processing C:\sync-tmp\Killjoys\Season 2\Killjoys - S02E10 - How to Kill Friends and Influence People.mkv.
2016-09-03 15:13:47 - mkvtomp4 - INFO - Reading video stream.
2016-09-03 15:13:48 - mkvtomp4 - INFO - Video codec detected: h264.
2016-09-03 15:13:48 - mkvtomp4 - INFO - Pix Fmt: yuv420p.
2016-09-03 15:13:48 - mkvtomp4 - INFO - Reading audio streams.
2016-09-03 15:13:48 - mkvtomp4 - INFO - Audio detected for stream #1: ac3 [und].
2016-09-03 15:13:48 - mkvtomp4 - INFO - Creating audio stream 0 from source audio stream 1 [iOS-audio].
2016-09-03 15:13:48 - mkvtomp4 - INFO - Audio track is number 0 setting disposition to default
2016-09-03 15:13:48 - mkvtomp4 - INFO - Creating audio stream 1 from source stream 1.
2016-09-03 15:13:48 - mkvtomp4 - INFO - Audio Track is number 1 setting disposition to none
2016-09-03 15:13:48 - mkvtomp4 - INFO - Reading subtitle streams.
2016-09-03 15:13:48 - mkvtomp4 - ERROR - No valid subtitle language specified, cannot download subtitles.
2016-09-03 15:13:48 - mkvtomp4 - INFO - Starting conversion.
2016-09-03 15:17:02 - mkvtomp4 - INFO - c:\PostProcess\Killjoys - S02E10 - How to Kill Friends and Influence People.mp4 created.
2016-09-03 15:17:03 - SonarrPostConversion - INFO - Tagging C:\sync-tmp\Killjoys\Season 2\Killjoys - S02E10 - How to Kill Friends and Influence People.mkv with ID 281534 season 2 episode 10.
2016-09-03 15:17:09 - tvdb_mp4 - INFO - Tagging file: c:\PostProcess\Killjoys - S02E10 - How to Kill Friends and Influence People.mp4.
2016-09-03 15:17:09 - tvdb_mp4 - INFO - Trying to write tags.
2016-09-03 15:17:09 - tvdb_mp4 - INFO - Tags written successfully.
2016-09-03 15:17:10 - mkvtomp4 - INFO - Relocating MOOV atom to start of file.
2016-09-03 15:17:10 - qtfaststart - INFO - Removing free atom at 32 (8 bytes)
2016-09-03 15:17:10 - qtfaststart - INFO - Patching stco with 58104 entries
2016-09-03 15:17:10 - qtfaststart - INFO - Patching stco with 95942 entries
2016-09-03 15:17:10 - qtfaststart - INFO - Patching stco with 75963 entries
2016-09-03 15:17:10 - qtfaststart - INFO - Writing output...
2016-09-03 15:17:30 - mkvtomp4 - INFO - c:\PostProcess\Killjoys - S02E10 - How to Kill Friends and Influence People.mp4 moved to c:\sync-tmp.
2016-09-03 15:17:32 - SonarrPostConversion - ERROR - Unable to refresh plex, check your settings.
Traceback (most recent call last):
  File "C:\MP4Automator\autoprocess\plex.py", line 38, in refreshPlex
    xml_sections = minidom.parse(urlopen(base_url))
  File "C:\Python27\lib\urllib.py", line 87, in urlopen
    return opener.open(url)
  File "C:\Python27\lib\urllib.py", line 213, in open
    return getattr(self, name)(url)
  File "C:\Python27\lib\urllib.py", line 350, in open_http
    h.endheaders(data)
  File "C:\Python27\lib\httplib.py", line 1053, in endheaders
    self._send_output(message_body)
  File "C:\Python27\lib\httplib.py", line 897, in _send_output
    self.send(msg)
  File "C:\Python27\lib\httplib.py", line 859, in send
    self.connect()
  File "C:\Python27\lib\httplib.py", line 836, in connect
    self.timeout, self.source_address)
  File "C:\Python27\lib\socket.py", line 575, in create_connection
    raise err
IOError: [Errno socket error] [Errno 10061] No connection could be made because the target machine actively refused it

The error at the end is just for an attempt to update Plex, which I don't have. I've since set refresh to False for Plex and it doesn't happen anymore.

As I said, other files seem to have been fine since, so maybe it was just a weird aberration - I'll keep an eye on it, particularly for the affected shows. Ironically, it was the affected shows (Killjoys and Dark Matter) that were the main reason for trying to convert to mp4 in the first place. It's those mkvs that tend to cause buffering problems in Kodi. I've never seen them cause problems in VLC on my Mac though.

I'll try to get a small chunk of that file that shows the problem and share it with you somehow.

Thanks
